modal_proto/api.proto

```diff
@@ -1433,6 +1438,7 @@ service ModalClient {
   // Volumes
   rpc VolumeCreate(VolumeCreateRequest) returns (VolumeCreateResponse);
   rpc VolumeCommit(VolumeCommitRequest) returns (google.protobuf.Empty);
+  rpc VolumeGetFile(VolumeGetFileRequest) returns (stream google.protobuf.BytesValue);
```

Comment:
Do you foresee any issues with streaming GiB or TiB worth of data here? Just curious if there's a reason we didn't take the same approach for `SharedVolumeGetFile` :thinking: 

For really large files the stream is liable to be interrupted before the download completes? E.g. we deploy a change and churn server on the backend while the download is in progress.

Reply:
Tangentially related: maybe we can have this return indexed byte chunks so we can implement resume-ability?

Reply:
Yeh I definitely did 😅. I set a limit on the server-side PR, but it's a hand wavy limit. 

> For really large files the stream is liable to be interrupted before the download completes?

Yep it's a vulnerability for sure. The `SharedVolumeGetFile` also has a long blocking upload to S3 which can also get interrupted. So I think the difference is in how much data is pushed through the external server. The `SharedVolumeGetFile` mechanism adds an upload step to shift the long-running stream from external server to S3.

Reply:
Actually I think in the medium-term future we'd like to just directly return an pre-signed download link from store. We can't do this currently but will be able to soon-ish. 

This makes the `SharedVolumeGetFileResponse` API more appropriate for sure.

Reply:
> has a long blocking upload to S3 which can also get interrupted. So I think the difference is in how much data is pushed through the external server.

True that. At least we have have lots of bandwidth available internally in our backend, which might not be the case for clients. But yeah, downloading really large files will be painful medium-term :|